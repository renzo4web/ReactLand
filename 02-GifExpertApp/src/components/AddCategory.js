import { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategories }) => {
  const [inputVal, setInputVal] = useState("");

  const handleChange = ({ target }) => {
    const { value } = target;
    setInputVal(value.trim());

    console.log("calling handleChange");
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    if (!inputVal.trim().length) return;
    setCategories((prevCats) => [inputVal, ...prevCats]);
    setInputVal("");
    console.log("calling handleSubmit");
  };

  return (
    <form
      onSubmit={handleSubmit}
      className="m-4 flex items-center flex-wrap md:flex-row "
    >
      <input
        type="text"
        value={inputVal}
        onChange={handleChange}
        className="
        w-full
        md:w-60
        transition transform duration-500 ease-in-out 
        rounded
        p-3 focus:outline-none focus:ring-2 focus:ring-purple-600
        focus:scale-110"
      />

      <button
        type="submit"
        aria-label="add category"
        className="w-full rounded p-2 md:w-20 md:h-full md:ml-4 uppercase bg-purple-600 text-white
      transition hover:bg-purple-500 hover:scale-110"
      >
        search
      </button>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
