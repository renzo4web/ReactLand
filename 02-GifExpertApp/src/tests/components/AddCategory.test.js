import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("Test <AddCategory />", () => {
  const setCategories = jest.fn();
  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("snapshot test", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("test input.value state should change", () => {
    let input = wrapper.find("input");
    input.simulate("change", { target: { value: "test" } });

    input = wrapper.find("input");
    const { value } = input.props();
    expect(value).toBe("test");
  });

  test("should not submit a form when input is empty", () => {
    const form = wrapper.find("form");
    form.simulate("submit", { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test("should form be submited and clear input text", () => {
    let input = wrapper.find("input");
    input.simulate("change", { target: { value: "test" } });

    const form = wrapper.find("form");
    form.simulate("submit", { preventDefault() {} });

    const value = wrapper.find("input").prop("value");

    expect(setCategories).toHaveBeenCalled();
    expect(value).toBe("");
  });
});
