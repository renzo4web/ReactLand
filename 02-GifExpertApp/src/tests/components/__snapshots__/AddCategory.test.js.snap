// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AddCategory /> snapshot test 1`] = `
<form
  className="m-4 flex items-center flex-wrap md:flex-row "
  onSubmit={[Function]}
>
  <input
    className=" w-full md:w-60 transition transform duration-500 ease-in-out  rounded p-3 focus:outline-none focus:ring-2 focus:ring-purple-600 focus:scale-110"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    aria-label="add category"
    className="w-full rounded p-2 md:w-20 md:h-full md:ml-4 uppercase bg-purple-600 text-white transition hover:bg-purple-500 hover:scale-110"
    type="submit"
  >
    search
  </button>
</form>
`;
